specification/postgresql/Microsoft.OrionDB.Management/models.tsp

```diff
@@ -0,0 +1,547 @@
+  /** Key is invalid */
+  Invalid: "Invalid",
+}
+
+/**
+ * A value indicating whether Geo-Redundant backup is enabled on the cluster.
+ */
+union GeoRedundantBackupEnum {
+  string,
+  /** Geo-Redundant backup is enabled */
+  Enabled: "Enabled",
+  /** Geo-Redundant backup is disabled */
+  Disabled: "Disabled",
+}
+
+/**
+ * Public network access is enabled or not for Orion Db cluster.
+ */
+union ClusterPublicNetworkAccessState {
+  string,
+  /** Public network access is enabled */
+  Enabled: "Enabled",
+  /** Public network access is disabled */
+  Disabled: "Disabled",
+}
+
+/**
+ * The HA mode for the cluster.
+ */
+union HighAvailabilityMode {
+  string,
+  /** High availability is not enabled */
+  Disabled: "Disabled",
+  /** Zone redundant high availability is enabled*/
+  ZoneRedundant: "ZoneRedundant",
+  /** Same zone high availability is enabled*/
+  SameZone: "SameZone",
+}
+
+/**
+ * A state of a HA cluster that is visible to user.
+ */
+union ClusterHAState {
+  string,
+  /** High availability is not enabled */
+  NotEnabled: "NotEnabled",
+  /** Creating standby cluster */
+  CreatingStandby: "CreatingStandby",
+  /** Replicating data to standby */
+  ReplicatingData: "ReplicatingData",
+  /** Failing over to standby */
+  FailingOver: "FailingOver",
+  /** High availability is healthy */
+  Healthy: "Healthy",
+  /** Removing standby cluster */
+  RemovingStandby: "RemovingStandby",
+}
+
+/**
+ * Represents a Orion Db resource.
+ */
+model OrionDbResource is Azure.ResourceManager.TrackedResource<OrionDbResourceProperties> {
+  ...ResourceNameParameter<OrionDbResource, "orionDbClusterName", "clusters", "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*">;
+
+  /**
+   * The SKU (pricing tier) of the Orion DB default pool.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  defaultPoolSku?: Azure.ResourceManager.CommonTypes.Sku;
+
+  /**
+   * Describes the identity of the application.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  identity?: Azure.ResourceManager.CommonTypes.UserAssignedIdentity;
+}
+
+/**
+ * The properties of a OrionDbResource.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "For backward compatibility"
+model OrionDbResourceProperties {
+  /**
+   * The mode to create a new OrionDB cluster.
+   */
+  @visibility(Lifecycle.Create, Lifecycle.Update)
+  createMode?: CreateMode;
+
+  /**
+   * The administrator username for the Orion Db. Once created it cannot be changed..
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  administratorLogin?: string;
+
+  /**
+   * The administrator login password (required for Orion Db creation).
+   */
+  @visibility(Lifecycle.Create, Lifecycle.Update)
+  @secret
+  administratorLoginPassword?: string;
+
+  /**
+   * The version of the Orion DB.
```

Comment:
"The version of the Orion DB." => "The major version of the PostgreSQL engine hosted by the replicas of an OrionDB cluster."